src/cluster.py

```diff
@@ -249,28 +256,14 @@ def get_postgresql_version(self) -> str:
             if snp["name"] == charm_refresh.snap_name():
                 return snp["version"]
 
-    def cluster_status(
-        self, alternative_endpoints: list | None = None
-    ) -> list[ClusterMember] | None:
+    def cluster_status(self, alternative_endpoints: list | None = None) -> list[ClusterMember]:
         """Query the cluster status."""
         # Request info from cluster endpoint (which returns all members of the cluster).
-        # TODO we don't know the other cluster's ca
-        verify = self.verify if not alternative_endpoints else False
-        for attempt in Retrying(
-            stop=stop_after_attempt(
-                len(alternative_endpoints) if alternative_endpoints else len(self.peers_ips)
-            )
+        if response := self.parallel_patroni_get_request(
+            f"/{PATRONI_CLUSTER_STATUS_ENDPOINT}", alternative_endpoints
         ):
-            with attempt:
-                request_url = self._get_alternative_patroni_url(attempt, alternative_endpoints)
-
-                cluster_status = requests.get(
-                    f"{request_url}/{PATRONI_CLUSTER_STATUS_ENDPOINT}",
-                    verify=verify,
-                    timeout=API_REQUEST_TIMEOUT,
-                    auth=self._patroni_auth,
-                )
-                return cluster_status.json()["members"]
+            return response["members"]
+        raise RetryError(last_attempt=Exception("Unable to reach any units"))
```

Comment:
Most existing code should handle RetryErrors instead of empty lists.

Reply:
Great move!